src/plugins/data_source/server/cryptography/cryptography_client.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  buildClient,
+  CommitmentPolicy,
+  RawAesKeyringNode,
+  RawAesWrappingSuiteIdentifier,
+} from '@aws-crypto/client-node';
```

Comment:
Why did we use this library over node's built-in [Crypto](https://nodejs.org/docs/latest-v14.x/api/crypto.html)?

Reply:
This is related to the issue created by noCharger here:  [[MD] Research on encryption / decryption strategies · Issue #1756 · opensearch-project/OpenSearch-Dashboards](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1756) . in which will add more reasons about why pick AWS crypto solution.

To summary, The concern majorly to introduce aws sdk as a dependency while we could be using node native library. And the reason to still go with aws crypto is due to:
1. It provide an envelope encryption feature which is leveraged by current implementation: https://docs.aws.amazon.com/encryption-sdk/latest/developer-guide/concepts.html#envelope-encryption
2. For later plans integration with KMS could leverage the library 

Also one more question if we could add this dependency as a plugin dependency, but due to multidata source modifies base concepts including index pattern. The whole feature won't be easily implemented as a plugin.

Reply:
@kristenTian @noCharger Let's capture this on #1756 as well.

Reply:
@AMoo-Miki Thanks for approval, I am working on it.

@kristenTian  Thanks for summarize the offline discussion, I am gonna add more details in #1756